src/main/java/bio/terra/service/snapshotbuilder/utils/HierarchyQueryBuilder.java

```diff
@@ -92,73 +99,69 @@ public Query generateQuery(SnapshotBuilderDomainOption domainOption, int concept
   }
 
-  /**
-   * Given a concept ID, select all of its parent concept IDs. Note that a concept may be its own
-   * ancestor so that case must be excluded.
-   *
-   * <pre>
-   * {@code SELECT ancestor_concept_id FROM concept_ancestor
-   * WHERE descendant_concept_id = :conceptId: AND ancestor_concept_id != :conceptId:;}
-   * </pre>
-   */
-  private Query selectAllParents(int conceptId) {
-    var conceptAncestor =
-        TableVariable.forPrimary(TablePointer.fromTableName(ConceptAncestor.TABLE_NAME));
-    var conceptIdLiteral = new Literal(conceptId);
-    FieldVariable ancestorConceptId =
-        conceptAncestor.makeFieldVariable(ConceptAncestor.ANCESTOR_CONCEPT_ID);
-    return new Query(
-        List.of(ancestorConceptId),
-        List.of(conceptAncestor),
-        BooleanAndOrFilterVariable.and(
-            BinaryFilterVariable.equals(
-                conceptAncestor.makeFieldVariable(ConceptAncestor.DESCENDANT_CONCEPT_ID),
-                conceptIdLiteral),
-            BinaryFilterVariable.notEquals(ancestorConceptId, conceptIdLiteral)));
+  static SourceVariable joinToFilterConcepts(FieldVariable parentId, int conceptId) {
+    var conceptAncestorTable =
+        SourceVariable.forPrimary(TablePointer.fromTableName(ConceptAncestor.TABLE_NAME));
+    var ancestorConceptId =
+        conceptAncestorTable.makeFieldVariable(ConceptAncestor.ANCESTOR_CONCEPT_ID);
+    var descendantConceptId =
+        conceptAncestorTable.makeFieldVariable(ConceptAncestor.DESCENDANT_CONCEPT_ID);
+    var subquery =
+        new Query(
+            List.of(ancestorConceptId, descendantConceptId),
+            List.of(conceptAncestorTable),
+            BooleanAndOrFilterVariable.and(
+                BinaryFilterVariable.equals(descendantConceptId, new Literal(conceptId)),
+                BinaryFilterVariable.notEquals(ancestorConceptId, new Literal(conceptId))),
+            null);
+    var subQueryPointer = new SubQueryPointer(subquery, "join_filter");
+    return SourceVariable.forJoined(subQueryPointer, ConceptAncestor.ANCESTOR_CONCEPT_ID, parentId);
   }
 
-  /**
-   * Generate a subquery that returns true if the outerConcept has any children.
-   *
-   * <pre>{@code
-   * EXISTS (SELECT
-   *     1
-   *   FROM
-   *     concept_ancestor ca
-   *   JOIN
-   *     concept c2
-   *   ON
-   *     c2.concept_id = ca.descendant_concept_id
-   *     AND ca.ancestor_concept_id = c.concept_id
-   *     AND ca.descendant_concept_id != c.concept_id
-   *     AND c2.standard_concept = 'S') AS has_children
-   * }</pre>
-   */
-  static SelectExpression hasChildrenExpression(FieldVariable conceptId) {
-    var conceptAncestor =
-        TableVariable.forPrimary(TablePointer.fromTableName(ConceptAncestor.TABLE_NAME));
+  static FieldVariable selectHChildren(SourceVariable joinHasChildren) {
+    return joinHasChildren.makeFieldVariable(
+        ConceptAncestor.DESCENDANT_CONCEPT_ID, "COUNT", "has_children", "> 0", true);
```

Comment:
Does this work on azure as well?

Reply:
Good question - The "> 0" clause is only applied to the GCP version.

Reply:
The way this column is interpreted by our code is under our control. So instead of generating different SQL for gcp and azure here, we could also change the code that reads the SQL response.

For example in `AggregateBQQueryResultsUtils`:
```
        .hasChildren(row.get(QueryBuilderFactory.HAS_CHILDREN).getLongValue() > 0)
```